src/WarriorRogueMage/CharacterCreation/WRMChargenCommon.cs

```diff
@@ -50,5 +53,15 @@ public static string FormatToColumn(int columnWidth, string stringToFormat)
 
             return retval;
         }
+
+        public static T GetFirstPriorityMatch<T>(string userQuery, IEnumerable<T> collection) where T: INamed
+        {
+            return (from r in collection
+                    where r.Name.StartsWith(userQuery, StringComparison.OrdinalIgnoreCase)
+                    select r).FirstOrDefault() ??
```

Comment:
@JeremyTHolland I wanted to call out that I gave poor advice a few days ago about how to do the matching priorities, that I noticed later when I got to better testing out the code changes.  This here is a better way to first finish the first search for things that "start with" and _then_ if an only if that finished search failed to find one, we search for any that "contains" the text.

(This PR also genericizes it and applies it to some other char gen picker steps.)